staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -765,12 +765,6 @@ func (r *Request) request(fn func(*http.Request, *http.Response)) error {
 	}
 
 	// TODO: added to catch programmer errors (invoking operations with an object with an empty namespace)
-	if (r.verb == "GET" || r.verb == "PUT" || r.verb == "DELETE") && r.tenantSet && len(r.resourceName) > 0 && len(r.tenant) == 0 {
```

Comment:
please verify that this does not cause regression with manual tests such as creating tenant, create pod & deployment and then delete them.

Reply:
I tried and pasted the test result in description.

Reply:
makes sense.